boxsdk/network/default_network.py

```diff
@@ -140,30 +144,23 @@ class DefaultNetworkResponse(NetworkResponse):
     _COMMON_RESPONSE_FORMAT = '"%(method)s %(url)s" %(status_code)s %(content_length)s\n%(headers)s\n%(content)s\n'
     SUCCESSFUL_RESPONSE_FORMAT = f'\x1b[32m{_COMMON_RESPONSE_FORMAT}\x1b[0m'
     ERROR_RESPONSE_FORMAT = f'\x1b[31m{_COMMON_RESPONSE_FORMAT}\x1b[0m'
-    STREAM_CONTENT_NOT_LOGGED = '<File download contents unavailable for logging>'
+    CONTENT_NOT_LOGGED = '<No content or content unavailable for logging>'
 
-    def __init__(self, request_response: 'Response', access_token_used: str):
+    def __init__(self, request_response: 'Response', access_token_used: str, log_response_content: bool = True):
         self._logger = getLogger(__name__)
         self._request_response = request_response
         self._access_token_used = access_token_used
         self._did_log = False
-        if not self.ok:
-            self.log(can_safely_log_content=True)
+        self.log(can_safely_log_content=log_response_content)
```

Comment:
I would strongly discourage making this change (moving the logging to the response constructor). It requires the caller of every network call to decide whether or not to log, rather than getting it right 99% or more of the time (with the current implementation).

It's probably OK to still look for the `log_response_content` in the constructor like we're doing, but save it as an instance variable, and combine it with `can_safely_log` in the `log` method. This would allow callers to opt out of logging, but would still get it right automatically most of the time.

Reply:
`It requires the caller of every network call to decide whether or not to log` - this is not true.
I used already existing parameter `expect_json_response`, which is being used each time when we aren't expecting json response (No Content response or bytes, e.g. for downloading files) and assign it to `log_response_content`. (log_response_content=request.expect_json_response). In request object the default value of `expect_json_response` is set True. Therefore if no `expect_json_response` specified in request() call content will be logged - don't need to specify anything everything is done automatically.

`It's probably OK to still look for the log_response_content in the constructor like we're doing, but save it as an instance variable, and combine it with can_safely_log in the log method. This would allow callers to opt out of logging, but would still get it right automatically most of the time.` - I don't see why this is wrong. When logging enabled I think we want log every network response so placing it in constructor makes sure that every one is logged (in opposition to previous solution when we logged only when accessing content or json, etc.). At this point we already know if we want to log response or not (We want to log messages only with json response) so why not put it there.

Reply:
`The SDK already doesn't log the content of downloaded files (as long as the developer uses the methods intended for downloading files)` but if they want to get the content of the file `file_content = client.file(file_id).content()` to do some operation on bytes first and then save it for file, then it is not possible to disable logging in current implementation.

Reply:
@Jeff-Meadows have above comments answered all your concerns? If no please let me know, otherwise I'll merge it tomorrow.